plugins/hypemachine/integration.js

```diff
@@ -0,0 +1,133 @@
+function update() {
+    return {
+        "playbackStatus": getPlaybackStatus(),
+        "canSeek": false,
+        "canGoNext": hasNext(),
+        "canGoPrevious": hasPrevious(),
+        "canAddToFavorites": true,
+        "volume": 1,
+        "duration": readTime("player-time-total"),
+        "position": readTime("player-time-position"),
+        "songId": getSongId(),
+        "songTitle": getSongTitle(),    
+        "artistName": getArtistName(),
+        "albumTitle": getAlbumTitle(),
+        "artUrl": getAlbumArt(),
+        "isFavorite": isFavorite()
+    };
+}
+
+function isFavorite() {
+    return hasClass(document.getElementById("playerFav"), "fav-on");
+}
+
+function hasNext() {
+
+    return true;
+}
+
+function hasPrevious() {
+    return true;
+}
+
+function hasClass(elem, className) {
+	try {
+		return elem.classList.contains(className);
+	}
+	catch (e) {
+        return false
+    }
+}
+
+function getPlaybackStatus() {
+    if (hasClass(document.getElementById("playerPlay"), "pause")) {
+        return mellowplayer.PlaybackStatus.PLAYING;
+    }
+    else {
+        return mellowplayer.PlaybackStatus.STOPPED;
+    }
+}
+
+function clickButton(buttonName) {
+    document.getElementById(buttonName).click();
+}
+
+function play() {
+    clickButton("playerPlay");
+}
+
+function pause() {
+    play();
+}
+
+function goNext() {
+    clickButton("playerNext");
+}
+
+function goPrevious() {
+    clickButton("playerPrev");
+}
+
+function setVolume(volume) {
+    // not supported
+}
+
+function addToFavorites() {
+    clickButton("playerFav");
+}
+
+function removeFromFavorites() {
+	clickButton("playerFav");
+}
+
+function seekToPosition(position) {
+	// not supported
+}
+
+function readTime(elementClassName) {
```

Comment:
MellowPlayer provide you with an utility function ``toSeconds`` that do exatcly what readTime does (see http://mellowplayer.readthedocs.io/en/latest/developers/plugins.html#utility-functions). Please use it instead.

Reply:
What about this comment?

Reply:
Now that I've had a look at your commit it looks good.